### PR TITLE
Allow Mentors to remove select days from small groups

### LINKED ISSUE #413: DCC 232 Small Group: Extraneous Number of Attendence Days
We have 11 reported days of attendance, but only 7 are valid. The days that are valid are: Feb 23, Mar 01, Mar 08, Mar 22, Mar 29, Apr 05, and Apr 12

### PR DESCRIPTION
**Pull request by agundy**
Fixes #413

### CODE REVIEW

**Comment by kmcnellis** [comment]
Can you add a count of how many people submitted? Otherwise it is just a guess if you will hurt people by deleting a code

**Comment by kmcnellis** [comment]
You should also be deleting the corresponding attendance objects

**Comment by agundy** [comment]
Updated to delete the attendance objects!

**Comment by agundy** [comment]
Fixed dem merge conflicts

File: server/api/smallgroup/smallgroup.controller.js [hunk 1]
```diff
@@ -114,6 +114,21 @@ exports.daycode = function(req, res){
...
+// router.delete('/:id/day/:dayCode', auth.hasRole('mentor'), controller.deleteDay);
+exports.deleteDay = function(req, res){
+    var dayCode = req.params.dayCode;
+    var smallGroupId = req.params.id;
+    SmallGroup.findOneAndUpdate({_id: smallGroupId}, {
+        $pull: { dayCodes: {code : dayCode }}
+    }, function(err, smallgroup){
+        if (err) return handleError(res, err);
+
+        return res.status(200).json(smallgroup);
```
Comment (kmcnellis) [inline server/api/smallgroup/smallgroup.controller.js]: This still does not delete any users' attendance who submitted that code

**Comment by kmcnellis** [comment]
Created #481 and #482 to continue this

**Pull request merged**